conanfile.py

```diff
@@ -44,27 +41,42 @@ def requirements(self):
             self.requires("gtest/1.14.0")
             self.requires("boost/1.84.0")
         if self.options.build_unbundled: #each componenet is built independently 
-            self.requires("up-cpp/0.1.1-dev")
-            if self.options.zenoh_package:
-                self.requires("zenohc/cci.20240213")
-        if self.options.build_cross_compiling :
+            self.requires("up-cpp/x.y.z")
             self.requires("zenohc/cci.20240213")
 
     def generate(self):
         tc = CMakeToolchain(self)
         tc.variables["BUILD_TESTING"] = self.options.build_testing
-        tc.variables["CROSS_COMPILE"] = self.options.build_cross_compiling
         tc.generate()
 
     def build(self):
+        if os.environ.get("BUILD_DIR_FULL_PATH") is not None:
+            return
         cmake = CMake(self)
         cmake.configure()
         cmake.build()
 
     def package(self):
-        cmake = CMake(self)
-        cmake.install()
+        if os.environ.get("BUILD_DIR_FULL_PATH") is not None:
+            self.output.info("BUILD_DIR_FULL_PATH: " + os.environ.get("BUILD_DIR_FULL_PATH"))
+            build_folder_path = os.environ.get("BUILD_DIR_FULL_PATH")
+            self.output.info("BUILD_DIR_FULL_PATH: " + build_folder_path)
+            self.copy(pattern='*', dst='.', src=('%s/install') % (build_folder_path), symlinks=True, ignore_case=False)
+        else:
+            cmake = CMake(self)
+            cmake.install()
 
     def package_info(self):
-        self.cpp_info.libs = ["up-client-zenoh-cpp"]
+        if os.environ.get("BUILD_DIR_FULL_PATH") is not None:
+            self.cpp_info.set_property("cmake_file_name", "up-client-zenoh-cpp")
+            self.cpp_info.set_property("cmake_target_name", "up-client-zenoh-cpp::up-client-zenoh-cpp")
+            self.cpp_info.set_property("pkg_config_name", "up-client-zenoh-cpp")
+            self.cpp_info.libs = collect_libs(self)
+
+            self.cpp_info.requires = ["up-cpp::up-cpp", "zenohc::zenohc", "spdlog::spdlog", "protobuf::protobuf"]
+
+            self.cpp_info.names["cmake_find_package"] = "up-client-zenoh-cpp"
+            self.cpp_info.names["cmake_find_package_multi"] = "up-client-zenoh-cpp"
+        else:
+            self.cpp_info.libs = ["up-client-zenoh-cpp"]
```

Comment:
This is a pretty significant change. Can you provide more of a description of what this does and why it is needed?